### DISCUSSION TITLE
Inner join on one-to-many relation returns data in String instead of []interface{}

### DISCUSSION THREAD

**Discussion by CRSylar** [jackc/pgx#2070] [Q&A] [answered]
Hi, i've some problem performing a query.

What i want to achieve is fairy simple, i need to Select an user based on its ID and join this result with a couple of other tables that are in a one-to-many relation with said user.

the Go structs for the user and the other tables are like this:

type User struct {
 Id  string
 jobs []Job
 achievements []achievement
....
}

type Job struct {
 Id string
 User_id string
 JobRole string
 StartDate time.Time
 EndDate time.Time
}

type Achievement Struct {
 Id string
 User_id string
 achievement string
 obtained_at time.Time
}

the Query im performing is this:
SELECT u.Id,
array_agg(DISTINCT a) as achievement,
array_agg(DISTINCT j) as job
FROM users u
INNER JOIN achievments a ON u.id = a.user_id
INNER JOIN Jobs j on u.id = j.user_id
WHERE u.id = $1
GROUP BY u.id,  a.id, j.id,
User.Id

since i'm sure to get at max 1 User i'm using a QueryRow, and it works fine, the problem is when i call the .Scan method passing 
&user.id, &user.achievements, &users.jobs

the Scan method error while trying to marshal into the achievement struct, after some debug i notice that the values returned from the query for achievements and job is not what i expected ( an Array of Interface) []interface{} but a String with the data like this:
"('idvalue', 'user_id_value', 'achievement_name', 'somedate')"

Can you help me understand this ? why i get back a string, i'm confused and after an entire afternoon of researches i don't know what to do.

**Comment by jackc** [comment, marked as answer]
I'm not sure what's going on with your code, but maybe the example in the docs could be helpful: https://pkg.go.dev/github.com/jackc/pgx/v5@v5.6.0/pgtype#hdr-Child_Records
